src/docs/wasm-compilation-pipeline.md

```diff
@@ -0,0 +1,57 @@
+
+## Debugging
+
+As mentioned earlier, TurboFan applies many optimizations, many of which involve re-ordering code, eliminating variables or even skipping whole sections of code. This means that if you want to set a breakpoint at a specific instruction, it might not be clear where program execution should actually stop. In other words, TurboFan code is not well suited for debugging. Therefore, when debugging is started by opening DevTools, all TurboFan code is replaced by Liftoff code again, as each WebAssembly instruction maps to exactly one section of machine code and all local and global variables are in-tact.
+
+## Profiling
+
+To make things a bit more confusing, within DevTools all code will get recompiled with TurboFan again when the Performance tab is opened and the "Record" button in clicked. The "Record" button starts performance profiling. Profiling the Liftoff code is not representative as it is only used while TurboFan isn’t finished and can be significantly slower than TurboFan’s output, which will be running for the vast majority of time.
+
+## Flags for experimentation
+
+For experimentation, V8 and Chrome can be configured to compile WebAssembly code only with Liftoff or only with TurboFan. It is even possible to experiment with lazy compilation, where functions only get compiled when they get called for the first time. The following flags enable these experimental modes:
+
+- Liftoff only:
+    - In V8, set the `--liftoff --no-wasm-tier-up` flags.
+    - In Chrome, disable [WebAssembly tiering](chrome://flags/#enable-webassembly-tiering) and enable [WebAssembly baseline compiler](chrome://flags/#enable-webassembly-baseline).
+
+- TurboFan only:
+    - In V8, set the `--no-liftoff --no-wasm-tier-up` flags.
+    - In Chrome, disable [WebAssembly tiering](chrome://flags/#enable-webassembly-tiering) and disable [WebAssembly baseline compiler](chrome://flags/#enable-webassembly-baseline).
+
+- Lazy compilation:
+    - Lazy compilation is a compilation mode where a function is only compiled when it is called for the first time. Similar to the production configuration the function is first compiled with Liftoff. After Liftoff compilation finishes, the function gets recompiled with TurboFan.
+    - In V8, set the --wasm-lazy-compilation flag.
+    - In Chrome, enable WebAssembly lazy compilation (chrome://flags/#enable-webassembly-lazy-compilation).
+
+## Compile time
+
+There are different ways to measure the compilation time of Liftoff and TurboFan. In the production configuration of V8, the compilation time of Liftoff can be measured from JavaScript by measuring the time it takes for `new WebAssembly.Module()` to finish, or the time it takes `WebAssembly.compile()` to resolve the promise. To measure the compilation time of TurboFan, one can do the same in a TurboFan-only configuration.
+
+![The trace for WebAssembly compilation in [Google Earth](https://earth.google.com/web)](/_img/wasm-compilation-pipeline/trace.svg)
```

Comment:
There seems to be some whitespace around the screenshot, in particular below it. Can you fix this?

Reply:
Done.